crates/node/src/command.rs

```diff
@@ -313,12 +277,27 @@ pub fn run() -> sc_cli::Result<()> {
                     Some((madara_path + "/p2p-key.ed25519").into());
             }
 
-            set_chain_spec(&mut cli)?;
+            if cli.run.run_cmd.shared_params.dev {
+                set_dev_environment(&mut cli);
+            }
+
+            if let Some(chain_spec_url) = cli.run.fetch_chain_spec.clone() {
```

Comment:
Do you have to clone? Cant `set_chain_spec` take a ref?

Reply:
It becomes conflicts with the mut cli there

Reply:
`set_chain_spec` is only used once, with an input that comes form the `cli` iteslf. 

So, `chain_spec_url` has no need to be passed as an argument.

I would change this func signatrue so it only takes `cli` as arg and does the if some check internaly